### PR TITLE
python: Add tests for container steps

### PR DESCRIPTION
**Pull request by yoff**
- constructors: list, tuple, set, dict
- methods:
  - general: copy, pop
  - list: append
  - set: add
  - dict: keys, values, items, get, popitem
- functions: sorted, reversed, iter, next

### CODE REVIEW

**Review by RasmusWL — CHANGES_REQUESTED** [review]
(I should have requested changes before :shrug:)

File: python/ql/test/experimental/dataflow/coverage/test_builtins.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    SINK(item_list[1][0]) #$ MISSING: flow="SOURCE, l:-5 -> item_list[1][0]"
+    SINK_F(item_list[1][1]) # expecting FP due to imprecise flow
+
+@expects(2)
+def test_dict_pop():
+    d = {'k': SOURCE}
+    v = d.pop("k")
+    SINK(v) #$ flow="SOURCE, l:-2 -> v"
+    v1 = d.pop("k", SOURCE)
+    SINK(v1) #$ flow="SOURCE, l:-4 -> v1"
```
[the remark this answers is not in the document]
Comment (yoff) [reply python/ql/test/experimental/dataflow/coverage/test_builtins.py:223]: Good catch, I meant for this to be -1 like your suggestion below, but making it more explicit by changing the key is a good idea. Also nice to document the spurious flow.

File: python/ql/test/experimental/dataflow/tainttracking/defaultAdditionalTaintStep/test_collections.py [hunk 2]
```diff
@@ -64,6 +72,29 @@ def test_access(x, y, z):
...
+        tainted_list[0], # $ tainted
+        tainted_list[x], # $ tainted
+        tainted_list[y:z], # $ tainted
+
+        sorted(tainted_list)[0], # $ tainted
+        reversed(tainted_list)[0], # $ tainted
+        iter(tainted_list), # $ tainted
+        next(iter(tainted_list)), # $ tainted
+        [i for i in tainted_list], # $ tainted
+        [tainted_list for _i in [1,2,3]], # $ MISSING: tainted
```
[the remark this answers is not in the document]
Comment (yoff) [reply python/ql/test/experimental/dataflow/tainttracking/defaultAdditionalTaintStep/test_collections.py:88]: Perhaps we should have also `[TAINTED_STRING for i in [1]]` as a more direct example.

Comment (yoff) [inline python/ql/test/experimental/dataflow/tainttracking/defaultAdditionalTaintStep/test_collections.py:88]:
```suggestion
        [tainted_list for _i in [1,2,3]], # $ MISSING: tainted
        [TAINTED_STRING for i in [1]], # $ MISSING: tainted
```

**Comment by yoff** [comment]
Slightly optimistic with the re-request, there 😅 Should be good now 🤞

**Review by RasmusWL — APPROVED** [review]

**Pull request merged**